Bump31

thebitpit opened the pull request:
Bump MythTV and MythPlugins to version 3.1
Ready for review

thesamesam commented on media-plugins/mythplugins/mythplugins-31.0.ebuild:
```diff
@@ -0,0 +1,165 @@
+Additional plugins may be installed with USE flags mytharchive mythgame mythzmserver mythzoneminder
+"
+
+src_prepare() {
```
Not needed.

thebitpit replied:
I wondered about that - Removed

thesamesam commented on media-plugins/mythplugins/mythplugins-31.0.ebuild:
```diff
@@ -0,0 +1,165 @@
+	dev-qt/qtsql:5
+	media-libs/freetype:2
+	media-libs/libpng:=
+	sys-apps/util-linux
```
util-linux is part of @system, so don't think it's needed. [See](https://devmanual.gentoo.org/general-concepts/dependencies/) "implicit system dependency".

thebitpit replied:
Removed

thesamesam commented on media-plugins/mythplugins/mythplugins-31.0.ebuild:
```diff
@@ -0,0 +1,165 @@
+		sys-libs/libraw1394
+	)
+	libass? ( media-libs/libass:= )
+	=media-tv/mythtv-${PV}*[alsa?,cdda?,cdr?,exif?,fftw?,ieee1394?,libass?,opengl?,raw?,xml?,xvid]
```
I wonder if a ~ would suffice here.

thebitpit replied:
Older versions of MythTV often are used for years because each major version generates a new database schema. Both `mythfrontend` and `mythbackend` must use the same database schema to work together. As a result releases are not frequent. 
I may need to make a patch version (i.e. mythtv-31.0_p20nnnnnn.ebuild) from the fixes/XX branch to fix a bug in MythTV.

Now that I have tried to justify my choice I noticed that there is an error in my logic. The database version only changes on major versions. It remains the same for minor versions and the fixes/XX branch. The correct dependency is: `=media-tv/mythtv-${MAJOR_PV}*...` where `MAJOR_PV=31` is near the beginning of the ebuild.

thesamesam replied:
Okay, I follow you. But ~ would allow you to make revisions like media-tv/mythtv-SOMEVERSION-r1 and so on, without needing to update all the dependent ebuilds. Maybe the * already has this effect here, though?

thebitpit replied:
As I understand it, the `*` allows anything after mythplugins-31 to match as `=`. The documentation specifically notes that `=` is required.

thesamesam commented on media-plugins/mythplugins/mythplugins-31.0.ebuild:
```diff
@@ -0,0 +1,165 @@
+		dev-perl/DateTime-Format-ISO8601
+		dev-perl/SOAP-Lite
+		dev-perl/JSON
+		=media-tv/mythtv-${PV}*[perl]
```
~?

thebitpit replied:
See above

thesamesam commented on media-plugins/mythplugins/mythplugins-31.0.ebuild:
```diff
@@ -0,0 +1,165 @@
+		dev-python/oauth
+		dev-python/pycurl
+		dev-python/urllib3
+		=media-tv/mythtv-${PV}*[python]
```
~?

thebitpit replied:
See above

thesamesam commented on media-plugins/mythplugins/mythplugins-31.0.ebuild:
```diff
@@ -0,0 +1,165 @@
+		app-cdr/dvd+rw-tools
+		dev-python/pillow
+		dev-python/mysqlclient
+		=media-tv/mythtv-${PV}*[python]
```
~?

thebitpit replied:
See above

mgorny commented:
@thesamesam is your review finished?

thebitpit commented:
> @thesamesam is your review finished?

Yes. I am emerging and running this version without problems.

thebitpit commented:
Ready for merging into tree -- I hope.

juippis commented on media-plugins/mythplugins/metadata.xml:
```diff
@@ -17,7 +17,7 @@
 		<flag name="mytharchive">Allows you to archive recorded content to CD or DVD</flag>
 		<flag name="mythbrowser">Adds a web browser within MythTV</flag>
-		<flag name="mythgallery">Adds a picture viewing gallery within MythTV</flag>
+		<flag name="mythgallery">Adds a picture viewing gallery within MythTV (removed in 31.0)</flag>
```
This additional comment is superfluous, it's not a correct place to write down changelog into.

thebitpit replied:
Comment removed

juippis commented on media-plugins/mythplugins/mythplugins-30.0-r2.ebuild:
```diff
@@ -16,7 +16,7 @@ HOMEPAGE="https://www.mythtv.org"
 LICENSE="GPL-2+"
 KEYWORDS="~amd64 ~x86"
-SLOT="0"
+SLOT="0/${PV}"
```
This would require a revbump.

thebitpit replied:
Remove.

juippis commented on media-plugins/mythplugins/mythplugins-31.0.ebuild:
```diff
@@ -0,0 +1,163 @@
+# Change this when bumping to a new major version.
+MAJOR_PV=31
+
+PYTHON_COMPAT=( python2_7 python3_{6,7} )
```
Aaaany chance you'd remove python2_7 altogether?

thebitpit replied:
Upstream supports python2_7 and one plugin has not been converted yet.

juippis commented on media-plugins/mythplugins/mythplugins-31.0.ebuild:
```diff
@@ -0,0 +1,163 @@
+inherit python-single-r1 readme.gentoo-r1
+
+DESCRIPTION="Official MythTV plugins"
+HOMEPAGE="https://www.mythtv.org"
```
Please also add the Github page here.

thebitpit replied:
Done

juippis commented on media-tv/mythtv/mythtv-31.0.ebuild:
```diff
@@ -0,0 +1,431 @@
+
+EAPI=7
+
+PYTHON_COMPAT=( python2_7 python3_{6,7} )
```
Any chance to get rid of python2_7 here also?

thebitpit replied:
Upstream supports python2_7 and one plugin has not been converted yet.

juippis commented on media-tv/mythtv/mythtv-31.0.ebuild:
```diff
@@ -0,0 +1,431 @@
+MYTHTV_BRANCH="fixes/${P%.*}"
+
+DESCRIPTION="Open Source DVR and media center hub"
+HOMEPAGE="https://www.mythtv.org"
```
Please also add the Github page.

thebitpit replied:
Done

juippis commented on media-tv/mythtv/mythtv-31.0.ebuild:
```diff
@@ -0,0 +1,431 @@
+	media-libs/libsamplerate
+	media-libs/taglib
+	sys-libs/zlib
+
```
Extra newline.

thebitpit replied:
Removed

juippis commented on media-tv/mythtv/mythtv-31.0.ebuild:
```diff
@@ -0,0 +1,431 @@
+		x11-wm/evilwm
+	)
+	bluray? (
+		media-libs/libbluray:=[java?]
+		dev-libs/libcdio:=
```
Sort while at it :)

thebitpit replied:
Sorted

juippis commented on media-tv/mythtv/mythtv-31.0.ebuild:
```diff
@@ -0,0 +1,431 @@
+	systemd? ( sys-apps/systemd:= )
+	vaapi? ( x11-libs/libva:=[opengl] )
+	vdpau? ( x11-libs/libvdpau )
+	vpx? ( <media-libs/libvpx-1.8.0:= )
```
Are you sure about this libvpx requirement? This will bring a lot of problems for everyone.

thebitpit replied:
Restriction not needed - removed

juippis commented on media-tv/mythtv/mythtv-31.0.ebuild:
```diff
@@ -0,0 +1,431 @@
+		x11-misc/wmctrl:=
+	)
+	x265? (	media-libs/x265 )
+	xml? ( dev-libs/libxml2 )
```
Add slot operator, `:2` here.

thebitpit replied:
Done

juippis commented on media-tv/mythtv/mythtv-31.0.ebuild:
```diff
@@ -0,0 +1,431 @@
+	opengl? ( virtual/opengl )
+	python? ( ${PYTHON_DEPS} )
+"
+DEPEND="
```
None of the libraries in `RDEPEND` are needed during compile-time?

juippis commented on media-tv/mythtv/mythtv-31.0.ebuild:
```diff
@@ -0,0 +1,431 @@
+	fi
+
+	if use autostart; then
+		echo CONFIG_PROTECT=\"$(egethome mythtv)\" > "${T}"/95mythtv
+		doenvd "${T}"/95mythtv
```
You can do this without using `${T}` with
```
newenvd - 95mythtv <<- _EOF_
	CONFIG_PROTECT=\"$(egethome mythtv)\"
_EOF_
```

thebitpit replied:
Done as suggested.

juippis commented on media-tv/mythtv/mythtv-31.0.ebuild:
```diff
@@ -0,0 +1,431 @@
+
+	insinto /usr/share/mythtv/contrib
+	# Ensure we don't install scripts needing the perl bindings (bug #516968)
+	use perl || find contrib/ -name '*.pl' -exec rm -f {} \;
```
Open up properly, `if use; then` and add `|| die` to `find` command. Remove `-f`.

thebitpit replied:
Find failure is not an error.

juippis commented on media-tv/mythtv/mythtv-31.0.ebuild:
```diff
@@ -0,0 +1,431 @@
+	# Ensure we don't install scripts needing the perl bindings (bug #516968)
+	use perl || find contrib/ -name '*.pl' -exec rm -f {} \;
+	# Ensure we don't install scripts needing the python bindings (bug #516968)
+	use python || find contrib/ -name '*.py' -exec rm -f {} \;
```
Same here.

thebitpit replied:
Find failure is not an error, there may be no python scripts.

thebitpit commented:
It appears that upstream changed https://github.com/MythTV/mythtv/archive/v31.0.tar.gz file. I deleted my old /var/cache/distfiles/mythtv-31.0.tar.gz file and the manifest. I believe I fixed the unexpected change to the source tarball. It compiles for me.

juippis commented on media-tv/mythtv/mythtv-31.0.ebuild:
```diff
@@ -0,0 +1,435 @@
+	insinto /usr/share/mythtv/contrib
+	# Ensure we don't install scripts needing the perl bindings (bug #516968) Finding none is OK.
+	if use perl; then
+		find contrib/ -name '*.pl' -exec rm {} \;
```
The `|| die` is also if `find` executable cannot be found or something. You need to be aware when this files exists and when not, but I believe the USE flag is used for that.

thebitpit replied:
The files being removed are user contributed scripts that are sometimes helpful when someone screws up their system. They are never required for the operation of MythTV. Installing the system should not fail even it "find" is not available. The third "find" does fail if scripts that remain can not be made executable if enabled by USE flag.

thebitpit commented:
Problem with Manifest?

The Manifest file in /usr/local/portage/media-tv/mythtv looks very different from the one in my github clone ~/repo/gentoo/media-tv/mythtv (same as Pull Request) and it does not build. Only 3 DIST files are listed, none of the other files in the directory are listed.

When I run `repoman manifest` in directory /usr/local/portage/media-tv/mythtv it produces a complete Manifest that builds. All the files in the directory are listed including 3 DIST files.

Is this normal operation? Do I copy the Manifest that works back into my local clone and commit that? Am I using repoman improperly? Something else?

I was forced to delete /var/cache/distfiles/mythtv-31.0.tar.gz because upstream changed the tagged version of the tarball after my first commit. (I saved the original -- they are indeed different) The Manifest in the PR matches the new upstream now.

To insure download from github I have in my /etc/portage/make.conf file:
GENTOO_MIRRORS=""

This works for me:
```
saxophone ~ #   USE="X alsa cdda cdr dvd exif lame opengl wrapper x264 xml xvid (-altivec) -asi -autostart -bluray -cec -ceton -debug -dvb -egl -fftw -hdhomerun -hdpvr -ieee1394 -ivtv -jack -java -lcd -libass -lirc -nvdec -oss -perl -pulseaudio -python -raw -systemd -v4l -vaapi -vbox -vdpau -vpx -x265 -xmltv -zeroconf" INPUT_DEVICES="joystick" emerge -va media-tv/mythtv

These are the packages that would be merged, in order:

Calculating dependencies... done!

[ebuild   R   ~] media-tv/mythtv-31.0::localrepo  USE="X alsa cdda cdr dvd exif lame opengl wrapper x264 xml xvid (-altivec) -asi -autostart* -bluray* -cec -ceton -debug -dvb* -egl -fftw* -hdhomerun* -hdpvr -ieee1394 -ivtv -jack -java* -lcd -libass -lirc -nvdec -oss -perl* -pulseaudio* -python* -raw -systemd -v4l* -vaapi* -vbox -vdpau* -vpx* -x265* -xmltv -zeroconf" INPUT_DEVICES="joystick*" 0 KiB

Total: 1 package (1 reinstall), Size of downloads: 0 KiB

Would you like to merge these packages? [Yes/No] yes
>>> Verifying ebuild manifests
>>> Emerging (1 of 1) media-tv/mythtv-31.0::localrepo
>>> Installing (1 of 1) media-tv/mythtv-31.0::localrepo
>>> Jobs: 1 of 1 complete                           Load avg: 6.25, 7.31, 5.73
>>> Auto-cleaning packages...

>>> No outdated packages were found on your system.

 * GNU info directory index is up-to-date.

```

juippis commented:
What version of Qt libraries you're on? I've upgraded to latest. Seems like it's related to this bug, 
https://bugs.gentoo.org/726178

There are some closed bugs to give you an idea how to fix this issue.

thebitpit commented:
I use the stable version 5.14. Thanks for figuring out what is wrong.

thebitpit commented:
I created a patch "mythtv-31.0-Include_QPainterPath.patch" (not on one git hub yet). It applies and builds with Qt 5.14. I don't have Qt 5.15 installed on my system to test against.

This same patch works for versions 29.1 and 30.0. 
Should I name the patch "mythtv-29.1-Include_QPainterPath.patch"?
Should I use the same patch in all 3 ebuilds?
Should I add 2 more commits to this PR to revision bump 29.1 and 30.0 ebuilds?

thebitpit commented:
There should be 3 commits in this PR:
Bump mythtv to version 31.0
Bump mythplugins to version 31.0
Patch mythtv versions 29.1 and 30.0 to compile with Qt-5.15 - revision not bumped because it does not affect users who already managed to build it. Merge bump version 31.0 before this since the patch added there.
